table_foosball/database.rb

```diff
@@ -0,0 +1,117 @@
+require_relative "./text_tools"
+
+class Database
+  include TextTools
+
+  def initialize(name)
+    @name = name
+    @filename = "./#{name}.sdb"
+  end
+
+  def clear
+    File.truncate("./#{@name}", 0)
+  end
+
+  def save(objs)
+    file = File.open(@filename, "r+")    
+    file.rewind
```

Comment:
Exactly! Here's another nifty thing though, let's say you had one process write to this database with another one reading it at the same time. If you truncate the file, that operation doesn't apply instantly (atomicly). That means the file could potentially be corrupt (not valid) when the reader tries to access it. `File.rename` is atomic (instant), but truncating the file is not. That means the way (on Linux) to atomically overwrite a file is to write to a new file, and then replace it with the other one with a rename. Something like this:

```ruby
File.open("#{@filename}.new", "w") { |f| f.write(serialize_objects(objs)) }
File.rename("#{@filename}.new", @filename)
``` 

You don't actually have to do this, but thought you might find it interesting :)